src/Compilers/Test/Utilities/CSharp/CompilationTestUtils.cs

```diff
@@ -266,6 +266,17 @@ public static SemanticInfoSummary GetSemanticInfoSummary(this SemanticModel sema
                 summary.ImplicitConversion = semanticModel.GetConversion(initializer);
                 summary.MemberGroup = semanticModel.GetMemberGroup(initializer);
             }
+            else if (node is PatternSyntax pattern)
```

Comment:
>else if (node is PatternSyntax pattern) [](start = 12, length = 39)

Is this change related to the goal of this PR?

Reply:
Yes. One of my added tests is getting the semantic information off a PatternSyntax that's a function pointer type. It will crash without this change.

---
In reply to: [430759883](https://github.com/dotnet/roslyn/pull/44436#discussion_r430759883) [](ancestors = 430759883)